### ISSUE TITLE
Better error messaging/handling

### ISSUE THREAD

**Issue by tstephens-handy** [htmllint/htmllint#167]
I have run into this issue a few times, and hacked around it before, but would like to discover the root cause: `TypeError: Cannot read property 'test' of undefined`.  Ideally there would be a better error message, saying what rule it was trying to test when it found this error, or a way to show the stack trace.  This shows up in both grunt-htmllint and htmllint-cli, using htmllint version 0.5.0.

HTML file:
```
<div>
  <div class="header">
    <div class="title">{{title}}</div>
    <div class="sort-select"></div>
  </div>
  <div *ngFor="let item of list"></div>
</div>
```

.htmllintrc
```
{
  "attr-no-dup": true,
  "attr-name-ignore-regex": "^[\\*|\\[].*",
  "class-no-dup": true,
  "doctype-first": "smart",
  "doctype-html5": true,
  "focusable-tabindex-style": true,
  "head-req-title": true,
  "id-no-dup": true,
  "id-class-style": "none",
  "id-class-ignore-regex": "",
  "class-style": "dash",
  "img-req-src": true,
  "indent-style": "spaces",
  "indent-width": 2,
  "indent-width-cont": true,
  "title-no-dup": true,
  "attr-name-style": "camelCase"
}
```

**Comment by mlochbaum** [comment]
The fact that a Javascript error happens at all is a bug, but I agree that it would also be good to have better error reporting for these kinds of errors when they come up. I should have time to look at this over the weekend.

**Comment by mlochbaum** [comment]
The problem in this case is the line `"id-class-style": "none"`. There's no "none" format; use `false` (or any falsy value) to avoid any linting.

htmllint should check for invalid option values. I suppose it would be best to do this when the options object is defined or modified. I'll close this issue once I figure that out...

**Comment by mlochbaum** [comment]
Okay, htmllint now has an awesome option value parsing and option/rule dependency management system, which will probably be released in version 1.0.0. It generates an error whenever you pass an undefined value to an option, and sets that option's value to `false`. So that pretty much closes this issue.

The docs do say that "none" can be used for `class-style`. I've added it only as a valid value for `class-style` and will correct the wiki.
